zerver/views/__init__.py

```diff
@@ -91,9 +93,10 @@ def accounts_register(request):
         existing_user_profile = None
 
     validators.validate_email(email)
-
+    # If OPEN_REALM_CREATION is enabled all user sign ups should go through the
+    # sepcial URL with domain name so that REALM can be identified if multiple realms exist
     unique_open_realm = get_unique_open_realm()
-    if unique_open_realm:
+    if unique_open_realm is not None and not settings.OPEN_REALM_CREATION:
```

Comment:
Currently we identify whether it is user registration or Realm registrtion with the following 

```python
if realm is None and settings.OPEN_REALM_CREATION:
            domain = split_email_to_domain(email)
            do_create_realm(domain, form.cleaned_data['realm_name'])
            realm = get_realm(domain)
            set_default_streams(realm, ["social", "general", "zulip"]
```

If a unique_open_realm exist Realm value would be non zero and It would be treated as User registration. So I added the second condition `settings.OPEN_REALM_CREATION` to make sure that this does not happen. 

So if OPEN_REALM_CREATION is enabled users have to visit the register/domain URL for registering the account.